apps/omg/lib/omg/output/fungible_morevp_token.ex

```diff
@@ -14,26 +14,29 @@
 
 defmodule OMG.Output.FungibleMoreVPToken do
   @moduledoc """
-  Representation of the payment transaction output of a fungible token `currency`
+  Representation of the payment transaction output of a fungible token `currency`.
+  Fungible token outputs could have different `type_marker` but as long as they share the same behaviour
+  they are indistinguishable from code perspective.
   """
   alias OMG.Crypto
-  defstruct [:owner, :currency, :amount]
+  defstruct [:owner, :currency, :amount, :type_marker]
 
   @type t :: %__MODULE__{
           owner: Crypto.address_t(),
           currency: Crypto.address_t(),
-          amount: non_neg_integer()
+          amount: non_neg_integer(),
+          type_marker: binary()
         }
 
-  def from_db_value(%{owner: owner, currency: currency, amount: amount})
-      when is_binary(owner) and is_binary(currency) and is_integer(amount) do
-    %__MODULE__{owner: owner, currency: currency, amount: amount}
+  def from_db_value(%{owner: owner, currency: currency, amount: amount, type_marker: type_marker})
+      when is_binary(owner) and is_binary(currency) and is_integer(amount) and is_binary(type_marker) do
+    %__MODULE__{owner: owner, currency: currency, amount: amount, type_marker: type_marker}
   end
 
-  def reconstruct([owner, currency, amount]) do
+  def reconstruct([type_marker, owner, currency, amount]) do
```

Comment:
Probably not ready for this PR but I think this structure is no longer valid in the future version of contracts and instead is `[type_marker, [owner, currency, amount]]`